external/mlir-hal/lib/Dialect/MHAL/Transforms/Prefill.cpp

```diff
@@ -0,0 +1,140 @@
+
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/GPU/IR/GPUDialect.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/MHAL/IR/MHAL.h"
+#include "mlir/Dialect/MHAL/Transforms/Passes.h"
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/IRMapping.h"
+#include "mlir/IR/Location.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/IR/SymbolTable.h"
+#include "mlir/Support/LLVM.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/RegionUtils.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/ADT/TypeSwitch.h"
+#include "llvm/Support/Casting.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace mlir {
+namespace mhal {
+#define GEN_PASS_DEF_MHALPREFILLPASS
+#include "mlir/Dialect/MHAL/Transforms/Passes.h.inc"
+} // namespace mhal
+} // namespace mlir
+
+#define DEBUG_TYPE "mhal-prefill"
+
+using namespace mlir;
+namespace {
+class MHALPrefillPass
+    : public mhal::impl::MHALPrefillPassBase<MHALPrefillPass> {
+
+  Value createConstantOp(OpBuilder &builder, Type elementType,
+                         uint32_t initValue) {
+    Value constantOp;
+    auto loc = builder.getUnknownLoc();
+    if (llvm::isa<FloatType>(elementType)) {
```

Comment:
Yeah, this'll all go away if the prefill attribute has a type on it.

Reply:
yes, you're correct. The entire method went away.